api/goodreads.py

```diff
@@ -12,31 +11,45 @@ def goodreads(update, context):
     if not query:
         text = "*Usage:* `/gr {BOOK_NAME}`\n"\
                "*Example:* `/gr stoner`"
+        parse_mode = 'Markdown'
     else:
         response = get(f'https://www.goodreads.com/search.xml?key={config["GOODREADS_API_KEY"]}&q={query}')
         root = ET.fromstring(response.content)
 
         for node in root.iter('work'):
             if node is not None:
-                title = node.find('best_book').find('title').text
-                author = node.find('best_book').find('author').find('name').text
-                # rating = node.find('average_rating').text
-                cover = _cover_URL(title)
-                stars = f"⭐ {node.find('average_rating').text}"
-
-                text = f"<b>{title}</b>\n{author}\n{stars}"\
-                       f"<a href='{cover}'>&#8205;</a>"
-
+                id = node.find('best_book').find('id').text
+                text = make_result(id)
+                parse_mode = 'HTML'
                 break
             else:
-                text="No entry found."
+                text = "No entry found."
 
     message.reply_text(
         text=text,
-        parse_mode='HTML',
+        parse_mode=parse_mode,
         disable_web_page_preview=False,
     )
 
-def _cover_URL(title: str):
-    isbn = isbnlib.isbn_from_words(title)
-    return f"http://covers.openlibrary.org/b/isbn/{isbn}-L.jpg"
+
+def make_result(goodreads_id):
+    """ Search using Goodreads ID of item """
+    r = get(f'https://www.goodreads.com/book/show.xml?key={config["GOODREADS_API_KEY"]}&id={goodreads_id}')
+    root = ET.fromstring(r.content)
+
+    for node in root.iterfind('book'):
+        title = node.find('title').text
+        isbn = node.find('isbn13').text
+        year = node.find('publication_year').text
+        author = node.find('authors').find('author').find('name').text
```

Comment:
Could use `node.findtext` in all these and pass the path to it instead of a chain of finds
```suggestion
        author = node.findtext('authors/author/name')
```